File: ApplicationCore/Entities/Image.cs
using System.Collections.Generic;
using System.IO;

namespace ApplicationCore.Entities
{
    /// <summary>
    /// Info about stored Images
    /// </summary>
    /// <typeparam name="T"></typeparam>
    public abstract class Image<T>: FileInfo<T> where T: Entity
    {
        public Image(): base()
        {
        }
        public Image(string title, string contentType, int relatedId, string fullPath, Stream inputStream): base(title, contentType, relatedId, fullPath, inputStream)
        {
        }
        public Image(string title, string contentType, int relatedId, string fullPath) : this(title,  contentType, relatedId, fullPath, null)
        {
        }
        protected Image(Image<T> image): base(image)
        {
        }
    }
}

File: OctopusStore/ClientApp/src/app/view-models/image/image.ts
import { Entity } from "../entity";
import { FileInfo } from "../file-info/file-info";

export abstract class Image<T extends Entity> extends FileInfo<T> {

  public constructor(init?: Partial<Image<T>>) {
    super(init);
    Object.assign(this, init);
  }
}
